examples/motoko_examples/basic-dao/src/types.ts

```diff
@@ -0,0 +1,87 @@
+import { nat, Principal, Variant } from 'azle';
+
+export type Account = { owner: Principal; tokens: Tokens };
+export type Proposal = {
+  id: nat;
+  votes_no: Tokens;
+  voters: Principal[];
+  state: ProposalState;
+  timestamp: nat;
+  proposer: Principal;
+  votes_yes: Tokens;
+  payload: ProposalPayload;
+};
+export type ProposalPayload = {
+  method: string;
+  canister_id: Principal;
+  message: string;
+};
+
+export type ProposalState = Variant<{
+  // A failure occurred while executing the proposal
+  Failed?: string;
+  // The proposal is open for voting
+  Open?: null;
+  // The proposal is currently being executed
+  Executing?: null;
+  // Enough "no" votes have been cast to reject the proposal, and it will not be executed
+  Rejected?: null;
+  // The proposal has been successfully executed
+  Succeeded?: null;
+  // Enough "yes" votes have been cast to accept the proposal, and it will soon be executed
+  Accepted?: null;
+}>;
+export type Tokens = {
+  amount_e8s: nat; // TODO should be nat
+};
+// export type TransferArgs = { to: Principal; amount: Tokens };
+export type UpdateSystemParamsPayload = {
+  transfer_fee?: Tokens;
+  proposal_vote_threshold?: Tokens;
+  proposal_submission_deposit?: Tokens;
+};
+export type Vote = Variant<{
```

Comment:
I would think we should lowercase these, and also variants should have a `?` as part of the type, otherwise you would have to put both properties.